src/main/java/net/comments/sample/TrainingRoomLimited.java

```diff
@@ -7,26 +7,22 @@
 
 public class TrainingRoomLimited implements Room {
     private final int size;
-    private List<Person> enteredPerson;
-    private Teacher someteacher;
-
+    private final List<Person> enteredPerson;
 
     // Allows to create a room with a given size
 
-    public TrainingRoomLimited(int size, Teacher someTeacher) {
+    public TrainingRoomLimited(int size) {
         enteredPerson = new ArrayList<Person>(size);
         this.size = size;
-        this.someteacher = someTeacher;
     }
 
-    // Allows to come into the room. Only one teacher can be in the room.
+    // Allows to come into the room.
 
     public void accept(Person somePerson) {
 
-            if (enteredPerson.size() < this.size - 1) {
-                System.out.println("Student " + somePerson.getName() + " went into the room");
+        if (enteredPerson.size() < this.size) {
+            System.out.println(somePerson.getName() + " went into the room");
                 enteredPerson.add(somePerson);
-
             } else {
                 System.out.println("We don’t learn a student if it is not in the room.");
```

Comment:
Please use proper message. The room doesn't know something about learning.

Reply:
@extsoft What exact message should be? Something like this:
'@somePerson.getName' + 'cannot enter into the room due to all seats are occupied'

Reply:
@oteplova That's the great example.